### ISSUE TITLE
edX to Open edX App Data Migration Path

### ISSUE THREAD

**Issue by marcotuts** [openedx/openedx-app-ios#99]
**Goal:** 
edX mobile team to get a working internal build of the new open edx mobile applications to confirm feature gap list scope, theming capabilities, data migration needs, and more. 

**Potential scope draft:**
- edX mobile team to have a build of the new applications powered with their data
- Confirm migration / data path
- Confirm theming capabilities etc. 

**Outcomes:**
- Further debugging and triage process for gaps sooner rather than later 

....
Step 1 - Devop Alignment

**Comment by omerhabib26** [comment]
Hey @marcotuts , @volodymyr-chekyrta  considering the current implementation for the Android app, applying branding to the existing code can be achieved using native approaches. Our app currently supports two themes: LightMode and DarkMode. To accommodate multiple branding options, we can create various themes within the app and configure the following properties to match the desired theme:

- Color Scheme
- Fonts
- Image Resources

We have the flexibility to define as many themes as necessary and create a Dark Mode variant accordingly.
However, it's important to note that modifying the designated folders for fonts and image resources is a relatively straightforward task. On the other hand, adjusting the color scheme requires a more extensive overhaul of the existing implementation. Currently, the color scheme is defined alongside fonts and shapes in a theme file. To make the rebranding process more manageable, we propose relocating the color definitions to a separate constant file and referencing them from there.
Furthermore, there are multiple instances within the app where hard-coded values are used instead of the dynamic values provided by the theme. This could pose challenges during the rebranding process.
In addition to these changes, we should establish a structured approach for dimension values. Currently, there's no standardized practice for utilizing dynamic values. Each view depends on hard-coded dimension values, primarily for padding, margins, buttons, layout/view creation, and shapes like rounded corners, among others.

To provide a clear visual representation of the app after the rebranding process, we have updated the color scheme, fonts, and image resources following the approach mentioned above.

| Current Open edX App | Rebranding -> Color, fonts & Images  |
| --------------- | ------------ |
| <video  src="https://github.com/openedx/openedx-app-ios/assets/30689349/3036e566-e988-4d4e-9e8b-483a6af89bd3" height="400" />  |  <video  src="https://github.com/openedx/openedx-app-ios/assets/30689349/aa423d69-8548-4ad1-b0f0-a359457d7036" height="400" />  |

**Comment by moiz994** [comment]
Pasting iOS team's findings here as well.

### Multiple colors and images
App main colors and images are being used from the assets of the Core framework. At the moment, it will be a bit hard for someone to directly replace the colors of the app in the assets because maintaining the change set will require extra effort as it contains data for other images and icons. And some colors, like white, are hardcoded in the codebase.

Some images have a tinted color as well, like, for example, blue for the back arrow that is being used in the navbar. It will make changing a theme a bit harder and will require icon/image changes as well.


**Recommendation:**
It would be nice if another assets folder named Configurable Assets would be created in the Core framework, which would contain only the configurable items like colors, logos, auth colors, etc. All the colors should be used through the assets and theme classes.

The images that are being used in the navbar or bottom bar shouldn’t have any tinted color and the main theme color should be applied while using the icon so it matches the main theme color.

### Fonts
Although fonts are being used through the theme class, and the parsing of fonts is also being done in the theme class. A code-level change (theme class parsing update) will be required to use different fonts.

**Recommendation:**
The basic part of the fonts parsing can be moved to a separate key-value based file, like in a JSON file, where the names of the fonts can be changed as per the font file, like 
`"regular":"Inter-Regular"`. It will make the fonts changes easy and the code level changes free.

### Icons and splash screen
The asset folder of the target `OpenedX` can be replaced with a new one to change the icons and splash screen.

**Recommendation:**
There isn’t any change to make here; the existing architecture is configurable.

**Comment by moiz994** [comment]
**Update:** Axinite will work to incorporate the recommendations in the app. Tickets for this work to be created soon.

FYI: @touchapp @marcotuts @volodymyr-chekyrta @miankhalid please review the findings and share feedback. We can also discuss this in our Technical Check-in call on Monday.

**Comment by marcotuts** [comment]
At this point it seems some tickets / improvements are coming in relating to this work - Is there an edx app build that could eventually be shared out with some of the wg-mobile team members once it exists? It would be helpful to be able to see a built via testflight or whenever ready as a contrast with the existing open edx app to do a live comparison in the themes but also to compare against the edX app (once ready - don't know how close we are to this - @moiz994

**Comment by moiz994** [comment]
@marcotuts currently, we're just implementing the ability to update the theme of the app easily. We will work on the edX them of the app a little later in time but be happy to share that once it is ready.

**Comment by marcotuts** [comment]
Any update here on how close we think we are on this data migration / theme push?

**Comment by moiz994** [comment]
@miankhalid & @touchapp can share more details on the data migration.
As for the theme, I believe we've merged theming capabilities work.

**Comment by marcotuts** [comment]
This may be close to ready to mark complete? The testflight build and theme customization seem done but will keep this epic open until we have confirmed this is ok to close.

**Comment by moiz994** [comment]
@marcotuts yes, theme is almost done. It's finalized on iOS but now Android team needs to implement what was finalized on iOS.

Also, I still don't know what is the way forward in terms of the data migration. @miankhalid @touchapp need your help and updates on this.

**Comment by hassan6190** [comment]
@miankhalid @moiz994 @marcotuts Following are my findings on the data migration from edX to open edX

**Android:**

- On upgrading from edX to open edX (New App) the user gets logged out
- On upgrading all the downloaded videos get deleted
- On upgrading the app the Calendar is not synced with the course date
- Calendar Sync: When a user enables the calendar sync toggle then 2 events are created for the same course in the device’s calendar
- On upgrading the app the IAP, upgrade content is not yet available

**iOS:**

- On upgrading from edX to open edX (New App) the user gets logged out
- On upgrading all the downloaded videos get deleted
- On upgrading the app the Calendar is not synced with the course date
- Calendar Sync: When a user enables the calendar sync toggle then there are no duplicate events create
- On upgrading the app the IAP, upgrade content is not yet available

**Note:** I have also created the issue for this which is mentioned above

**Comment by marcotuts** [comment]
With the findings above, are we ok to close this epic out @miankhalid  / @moiz994 ? Or keep open a bit longer until the findings are explored / reviewed?
